pkg/client/unversioned/helper.go

```diff
@@ -310,12 +336,49 @@ func SetKubernetesDefaults(config *Config) error {
 	return nil
 }
 
+// SetKubernetesDefaults sets default values on the provided client config for accessing the
+// Kubernetes API or returns an error if any of the defaults are impossible or invalid.
+func SetDefaultsForGroup(group string, config *Config) error {
```

Comment:
This is SetKubernetesDefaultsForGroup, because it imposed defaults for API groups that are colocated with a Kube server.  For instance, prefix, user agent, and default group version are different for API groups on an Openshift client connecting to Openshift.  We can't resolve that until discovery populates defaultGroupVersion (which is hardcoded to the Kube scheme today).